### PR TITLE
Logcollector integration tests T0: Check log_format values

### PR DESCRIPTION
**Pull request by damarisg**
| Related issue   |
|  Closed  [1126 ](https://github.com/wazuh/wazuh-qa/issues/1226)             |

## Description

- [x] Create tests for valid and invalid values for the following localfile log_format values options:
        - Json
        - Snort-full
        - Squid
        - Syslog
        - Audit
        - mysql_log
        - Postgresql_log
        - nmapg
        - djb-multilog
        - multi-line:3
        - eventlog
        - eventchannel
        - iis

- [x] Testing and debugging process.
       - These tests will check that if you add a file with `valid` content is accepted and showed in `ossec.log.`
       - They will also make sure that the Wazuh manager shows an error in `ossec.log` when a file has `invalid` content.
       - Jenkins Results:  [Last Iteration](https://ci.wazuh.info/view/Tests/job/Test_integration/4094/).
            
        
## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [x] `provision_documentation.sh` generate the docs without errors.

## Documentation

imagen

imagen

imagen

### CODE REVIEW

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+no_restart_windows_after_configuration_set = True
+force_restart_after_restoring = True
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+
+local_internal_options = {'logcollector.vcheck_files': '1', 'logcollector.debug': '2', 'monitord.rotate_log': '0'}
+
+if sys.platform == 'win32':
+    location = r'C:\test.txt'
+#    iis_path = r'%SystemDrive%\inetpub\logs\LogFiles\W3SVC1\test.log'
```
Comment (spothound) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:32]: ?????

**Review by spothound — COMMENTED** [review]
JUst a few changes asked from my part.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+                lines = f.readlines()
+                for line in lines:
+                    log_callback = logcollector.callback_reading_file(conf['log_format'], line.rstrip('\n'))
+                    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                            error_message=logcollector.GENERIC_CALLBACK_ERROR_READING_FILE)
+
+    elif conf['log_format'] == 'iis':
+        log_callback = logcollector.callback_read_file(conf['location'])
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message=logcollector.GENERIC_CALLBACK_ERROR_READING_FILE)
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:361]:
```suggestion
    if conf['log_format'] in log_format_windows_print_analyzing_info:
        # For this case you're only covering iis logs
        if conf['log_format'] == 'iis':
            log_callback = logcollector.callback_read_file(conf['location'])
        
    elif conf['log_format'] in log_format_not_print_reading_info:
        # Logs format that only shows when a specific file is read.
        log_callback = logcollector.callback_read_file(conf['location'])

    elif conf['log_format'] == 'multi-line:3':
        # It is necessary to reorganize the content of the file to compare it
        # with the generated output of the 'multiline' format.

        msg = ""
        with open(location) as file:
            for line in file:
                msg += line.rstrip('\n')
                msg += ' '
            log_callback = logcollector.callback_reading_file(conf['log_format'], msg.rstrip(' '))
    else:
        # Verify that the content of the parsed file is equal to the output generated in the logs.
        with open(location) as file:
            for line in file:
                log_callback = logcollector.callback_reading_file(conf['log_format'], line.rstrip('\n'))
    
    wazuh_log_monitor.start(timeout=5, callback=log_callback,
                            error_message=logcollector.GENERIC_CALLBACK_ERROR_READING_FILE)
```

This `if` block is a bit messy. Give it another round and refactor it to make it easier to understand. Also, take into account that some cases are not covered yet.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+
+
+def analyzing_invalid_value(conf):
+    """
+    Checks for the error message that is shown when the record format type is valid but the content is invalid.
+    """
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:367]: Follow the standard for the docstings. Also, the `Args` and `Returns` fields are missing.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+
+    with open(conf['location']) as log:
+        line = log.readline()
+        log_callback = logcollector.callback_invalid_format_value(line.rstrip('\n'),
+                                                                  conf['log_format'], conf['location'])
+        return log_callback
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:373]:
```suggestion
    return log_callback
```

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+
+
+def check_log_format_value_invalid(conf):
+    """
+    Check if Wazuh fails because of content invalid log.
+
+    Raises:
+       TimeoutError: If error callback are not generated.
+   """
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:382]: Same here.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+        log_callback = analyzing_invalid_value(conf)
+        wazuh_log_monitor.start(timeout=5, callback=log_callback, error_message=logcollector.GENERIC_CALLBACK_ERROR)
+
+
+def check_log_format_values(conf):
+    """
+    Set of validations to follow to validate a certain content.
+    The file content could be valid or invalid.
+
+   """
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:396]: Same here.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 7]
```diff
@@ -0,0 +1,472 @@
...
+    if conf['valid_value']:
+        check_log_format_valid(conf)
+        modify_file(conf['location'], conf['log_format'], conf['valid_value'])
+        check_log_format_value_valid(conf)
+        file.remove_file(conf['location'])
+    else:
+        check_log_format_valid(conf)
+        modify_file(conf['location'], conf['log_format'], conf['valid_value'])
+        check_log_format_value_invalid(conf)
+        file.remove_file(conf['location'])
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:407]:
```suggestion
    check_log_format_valid(conf)    
    modify_file(conf['location'], conf['log_format'], conf['valid_value'])
    if conf['valid_value']:
        check_log_format_value_invalid(conf)
    else:
        check_log_format_value_valid(conf)
    file.remove_file(conf['location'])
```

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 8]
```diff
@@ -0,0 +1,472 @@
...
+def test_log_format(get_local_internal_options, get_configuration,
+                    configure_local_internal_options, configure_environment):
+    """
+    Check if Wazuh log format field of logcollector works properly.
+
+    Ensure Wazuh component fails in case of invalid content file and works properly in case of valid log format values.
+
+    Raises:
+        TimeoutError: If expected callbacks are not generated.
+    """
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:419]: Same here.

File: tests/integration/test_logcollector/test_log_format/test_log_format_values.py [hunk 9]
```diff
@@ -0,0 +1,472 @@
...
+
+            elif conf['log_format'] == 'nmapg':
+                create_file_location(nmap_log, conf['log_format'])
+                control_service('start', daemon=LOGCOLLECTOR_DAEMON)
+                check_log_format_values(conf)
+
+            else:
+                create_file_location(location, conf['log_format'])
+                control_service('start', daemon=LOGCOLLECTOR_DAEMON)
+                check_log_format_values(conf)
```
Comment (BraulioV) [inline tests/integration/test_logcollector/test_log_format/test_log_format_values.py:472]: This code-block can be simplified by unifying the common lines.

**Review by BraulioV — CHANGES_REQUESTED** [review]

**Review by BraulioV — APPROVED** [review]

**Review by spothound — APPROVED** [review]
lgtm

**Pull request merged**